client/store/userchallenge.js

```diff
@@ -20,6 +20,13 @@ export const updateUserchallenge = (userAnswer, userchallengeId, isSubmit) => di
     .then(userAnswer => dispatch(setUserchallenge(userAnswer)));
 };
 
+export const fetchUserchallengeById = id => dispatch => {
+  return axios
+    .get(`/api/userchallenges/${id}`)
+    .then(res => res.data)
+    .then(userchallenge => dispatch(setUserchallenge(userchallenge)));
+};
+
 export const fetchUserchallenge = (userId, challengeId) => dispatch => {
```

Comment:
Is your `fetchUserchallengeById` replacing `fetchUserchallenge`?  Seems like is doing the same thing, except your route does not include `userId`.  Your route is cleaner.

Reply:
Fetching user challenge takes a user id and a challenge id and fetches the active user challenge for that challenge. Mine is just grabbing a specific user challenge by it's particular id.